crates/ruma-common/src/push/condition.rs

```diff
@@ -590,7 +590,7 @@ mod tests {
         assert!(!"m".matches_word("[[:alpha:]]?"));
         assert!("[[:alpha:]]!".matches_word("[[:alpha:]]?"));
 
-        // From the spec: <https://spec.matrix.org/v1.4/client-server-api/#conditions-1>
+        // From the spec: <https://spec.matrix.org/latest/client-server-api/#conditions-1>
```

Comment:
Maybe the changes in this file should be reverted? There seems to be a high-ish risk that this link will refer to the wrong section in the future (though it seems correct right now).

Reply:
We could add a condition in the spec links check that fragments ending with a number shouldn't use the `latest` version?

Reply:
The ones without effectively with `-0`, so I don't think that's a good idea

Reply:
We could go for "if there's any duplicates that would be assigned `-num`, don't allow `/latest/` link" I guess. But I'd also be fine with not automating these exceptions.

Reply:
> The ones without effectively end with -0, so I don't think that's a good idea

No they don't, there's never a `-0` suffix. The first one doesn't have a number at the end, the second one has `-1` and so on.

> But I'd also be fine with not automating these exceptions.

I'd rather auomate it. Off the top of my head I remember having 5 invalid links ending with a number before I handled it, we only have 3 here :slightly_smiling_face: (I might be wrong though).

Reply:
Tbh I'd rather petition the spec to start using header ordering numbers (1.4.6. Etc) than go through with this change tbh, there's just too much risk of silent "breakage"

Reply:
> This also means that any link can point to the wrong fragment if the same ID is added before at some point.

I think that's what Jonathan meant with there being "effectively" a `-0` at the end for links that don't otherwise have a number.

> I'd rather automate it.

Do you think my suggestion is realistic there? I.e. raise an error on a link to `/latest/...#foo` if `foo-1` exists?

Reply:
> Do you think my suggestion is realistic there? I.e. raise an error on a link to /latest/...#foo if foo-1 exists?

Yes, no problem